Unify footer buttons on open_content and dashboard/link pages

jesusgonzalezrivera opened the pull request:
Unify cancel, save and publish/unpublish buttons on the different pages

## Testing instructions

1. Create an open content page
2. Check that only appears Cancel, Save and Publish/Unpublish buttons

---

1. Create a dashboard or link page
2. Check that only appears Cancel, Save and Publish/Unpublish buttons

## Pivotal Tracker

[https://www.pivotaltracker.com/story/show/169907748](https://www.pivotaltracker.com/story/show/169907748)

jesusgonzalezrivera committed:
Unify footer buttons on open_content and dashboard/link pages

clementprdhomme reviewed (changes requested):
Hey Jesús,

When you create an Open Content page, the “Content” step displays 3 buttons: “Cancel”, “Save” and “Publish”. When you click either “Save” or “Publish”, the user is brought to the next step: “Preview”. I don't think that's what the user expects. Instead, if they click one of the two buttons, they should leave the creation flow.

Given we want to maintain the “Preview” step, I think we should just display “Cancel” and “Continue” in the “Content” step, and “Cancel”, “Save” and “Publish” on the “Preview” one.

Secondly, I would also suggest the primary button is always the one on the far right. So, for all the non-final steps, the order would be: “Cancel”, secondary on the left and “Continue”, primary on the right. For the final steps, the order would be: “Cancel”, secondary on the left, “Publish”/“Unpublish”, secondary on the right, and “Save”, primary on the right.

Finally, I would like to suggest one last change for the keyboard users. When the focus is located within a form, if the user presses enter, the form is submitted. The submit action should be the same as clicking the (only) primary button. For non-final states, that would mean going to the next step (as by clicking on “Continue”) and for final states, that would mean saving the changes (as by clicking on “Save”).

I would like you to make sure the primary buttons are always buttons of type `submit` and the others of type `button`. Currently, it seems both the “Publish”/“Unpublish” buttons and “Save” are submit buttons, which is not ideal.

jesusgonzalezrivera committed:
Fix button on the footer for site page creation

- Only one submit button per page
- Primary button always on the right
- Only Cancel and Continue buttons on Content page
